### PR TITLE
chore(build-system-test): Merge web build system tests with canaries 

### PR DESCRIPTION
**Pull request by esauerbo**
#### Description of changes
- Copy `canary/environments` and `canary/e2e` to build-system-test folder (no changes made) 
- Update `reusable-build-system-test` workflow to set up credentials, pull environment, and run limited e2e (copied from `build-and-runtime-test` and lightly adapted)
- Add `mega-app-start` script to run the mega apps after they're built
- Minor updates to mega-app templates and build scripts to get everything working

Planning to remove `canary` folder and `build-and-runtime-test` in a followup in case we need to revert anything

#### Description of how you validated changes
Passing POC workflow that triggers web build system tests https://github.com/aws-amplify/amplify-ui/pull/5888/checks 

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build-system-tests/templates/components/react/vite/index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
+  </head>
+  <body>
+    <div id="root"></div>
+    <script type="module" src="/src/main.tsx"></script>
+    <script>
+      window.global = window;
```
Comment (reesscot) [inline build-system-tests/templates/components/react/vite/index.html:13]: This shouldn't be needed anymore with the latest version of aws-amplify

File: build-system-tests/templates/components/react/next/App.js [hunk 2]
```diff
@@ -1,78 +1,70 @@
...
+            <LocationSearch />
+          </MapView>
+        </main>
+      )}
+    </Authenticator>
   );
-}
+};
+
+export default dynamic(() => Promise.resolve(Home), {
```
Comment (reesscot) [inline build-system-tests/templates/components/react/next/App.js:68]: what does this do?

File: .github/workflows/build-system-test.yml [hunk 3]
```diff
@@ -14,6 +14,14 @@ jobs:
...
     with:
       dist-tag: latest
+    secrets:
+      AUTH_E2E_ROLE_ARN: ${{ secrets.AUTH_E2E_ROLE_ARN }}
+      DOMAIN: ${{ secrets.DOMAIN }}
+      PHONE_NUMBER: ${{ secrets.PHONE_NUMBER }}
+      USERNAME: ${{ secrets.USERNAME }}
+      NEW_PASSWORD: ${{ secrets.NEW_PASSWORD }}
+      VALID_PASSWORD: ${{ secrets.VALID_PASSWORD }}
+
```
Comment (calebpollman) [inline .github/workflows/build-system-test.yml:24]:
```suggestion
```

File: build-system-tests/templates/components/react/vite/App.tsx [hunk 4]
```diff
@@ -1,78 +1,62 @@
...
-        initialViewState={{
-          latitude: 37.8,
-          longitude: -122.4,
-          zoom: 14,
-        }}
-      >
-        <LocationSearch />
-      </MapView>
-    </>
+    <Authenticator>
```
Comment (calebpollman) [inline build-system-tests/templates/components/react/vite/App.tsx:13]: Afaict this is the same as the _next/App.tsx_ content. As a follow up can we put this code in a shared file and import in to both test files?

File: build-system-tests/templates/components/react/vite/vite.config.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import { NodeGlobalsPolyfillPlugin } from '@esbuild-plugins/node-globals-polyfill';
+import { NodeModulesPolyfillPlugin } from '@esbuild-plugins/node-modules-polyfill';
```
Comment (calebpollman) [inline build-system-tests/templates/components/react/vite/vite.config.js:4]: Do we still need these? Think they were required fo `aws-amplify@v5`

File: build-system-tests/templates/components/react/vite/index.html (hunk 1 above)
Comment (esauerbo) [reply build-system-tests/templates/components/react/vite/index.html:13]: I'm seeing this issue still with the ui-geo package

File: build-system-tests/templates/components/react/vite/App.tsx (hunk 4 above)
Comment (esauerbo) [reply build-system-tests/templates/components/react/vite/App.tsx:13]: Yeah makes sense will create a followup task for this

File: build-system-tests/templates/components/react/next/App.js (hunk 2 above)
Comment (esauerbo) [reply build-system-tests/templates/components/react/next/App.js]: This was to resolve [this](https://github.com/aws-amplify/amplify-ui/actions/runs/11281713272/job/31377552683) build error (and it's modeled after our liveness `next` examples) but I can't replicate the error anymore so took it out

File: build-system-tests/templates/components/react/vite/vite.config.js (hunk 5 above)
Comment (esauerbo) [reply build-system-tests/templates/components/react/vite/vite.config.js]: Oh yeah this actually isn't being used anywhere just forgot to take it out, removed

**Review by calebpollman — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
Thanks for getting this done!

**Pull request merged**